utils/getLinkData.js

```diff
@@ -0,0 +1,25 @@
+const { getLinkPreview } = require("link-preview-js");
+const { redisCache } = require("../config/cache");
+
+async function getLinkData(article_url) {
+  let results;
+  let isCached = false;
+
+  try {
+    const cacheResults = await redisCache.get(article_url);
+    if (cacheResults) {
+      isCached = true;
+      results = JSON.parse(cacheResults);
```

Comment:
I missed this while we were working on this together, but the we are not using the `isCached` variable, so you should remove it.